clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2829,24 +2829,27 @@ class SyclKernelIntHeaderCreator : public SyclKernelFieldHandler {
 } // namespace
 
 class SYCLKernelNameTypeVisitor
-    : public TypeVisitor<SYCLKernelNameTypeVisitor>,
-      public ConstTemplateArgumentVisitor<SYCLKernelNameTypeVisitor> {
+    : public TypeVisitor<SYCLKernelNameTypeVisitor, bool>,
+      public ConstTemplateArgumentVisitor<SYCLKernelNameTypeVisitor, bool> {
```

Comment:
Is there any value in making it return `bool`?
I mean, AFAIK visitors defined in SemaSYCL, like `SyclKernelUnionChecker` returning `bool` has value, because next visitors start execution only if previous succeeded. I've seen that base classes for clang visitors which we use here is simple and empty most of times, so it looks like there is some return type doesn't change anything and we don't use it anyway.

Reply:
The return type of both these Visitor classes had to be set to "bool", because the Visit methods implemented from both the classes return bool. 
`VisitTagDecl() `and `VisitEnumType()` are the 2 visit methods that emits the diagnostic. 
Initially, I had set the return type to bool to just the TypeVisitor,
but we invoke `VisitEnumType `from `ConstTemplateArgumentVisitor's `visit methods as well. 
Please take a look at : https://clang.llvm.org/doxygen/classclang_1_1TypeVisitor.html

Reply:
I see what you mean. We could get away with modifying the base Visitor classes and just invoke isValid().